pyqtorch/analog.py

```diff
@@ -665,15 +676,50 @@ def forward(
         Returns:
             The transformed state.
         """
+        if self.is_time_dependent and embedding is not None:
+            n_qubits = len(state.shape) - 1
+            batch_size = state.shape[-1]
+            tsave = torch.linspace(0, float(self.duration), self.steps)
+
+            def Ht(t: torch.Tensor) -> torch.Tensor:
+                """Accepts a value 't' for time
+                and returns a (2**n_qubits, 2**n_qubits) Hamiltonian.
+                """
+                # We use the origial embedded params and return a new dict
+                # where we reembedded all parameters depending on time with value 't'
+                reembedded_time_values = embedding.reembed_tparam(values, t)
+                return (
+                    self.generator[0]
+                    .tensor(reembedded_time_values, n_qubits)
+                    .squeeze(2)
+                )  # squeeze the batch dim and return shape (2**n_qubits, 2**n_qubits)
+
+            sol = sesolve(
+                Ht,  # returns a tensor of shape (2**n_qubits, 2**n_qubits)
+                torch.flatten(
+                    state, start_dim=0, end_dim=-2
+                ),  # reshape to (2**n_qubits, batch_size)
+                tsave,
+                self.solver_name,
+            )
 
-        evolved_op = self.tensor(values, None)
-        return apply_operator(
-            state=state,
-            operator=evolved_op,
-            qubits=self.qubit_support,
-            n_qubits=len(state.size()) - 1,
-            batch_size=evolved_op.shape[BATCH_DIM],
-        )
+            state = sol.states[
```

Comment:
Would it make sense to add a `final_state` method to `states` ? I think Pulser does it under the hood.